ci-operator/step-registry/ipi/conf/vsphere/dns/ipi-conf-vsphere-dns-commands.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+set -x
+
+cluster_profile=/var/run/secrets/ci.openshift.io/cluster-profile
+cluster_name=${NAMESPACE}-${JOB_NAME_HASH}
+base_domain=origin-ci-int-aws.dev.rhcloud.com
+cluster_domain="${cluster_name}.${base_domain}"
+
+export AWS_SHARED_CREDENTIALS_FILE=${cluster_profile}/.awscred 
+
+# Load array created in setup-vips:
+# 0: API
+# 1: Ingress
+# 2: DNS
+declare -a vips
+mapfile -t vips < "${SHARED_DIR}"/vips.txt
+
+hosted_zone_id="$(aws route53 list-hosted-zones-by-name \
+            --dns-name "${base_domain}" \
+            --query "HostedZones[? Config.PrivateZone != \`true\` && Name == \`${base_domain}.\`].Id" \
+            --output text)"
+
+
+echo "Creating DNS records..."
+cat > "${SHARED_DIR}"/dns-create.json <<EOF
+{
+"Comment": "Create public OpenShift DNS records for VSphere IPI CI install",
+"Changes": [{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "api.$cluster_domain.",
+      "Type": "A",
+      "TTL": 60,
+      "ResourceRecords": [{"Value": "${vips[0]}"}]
+      }
+    },{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "*.apps.$cluster_domain.",
+      "Type": "A",
+      "TTL": 60,
+      "ResourceRecords": [{"Value": "${vips[1]}"}]
+      }
+    },{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "$cluster_domain.",
+      "Type": "NS",
+      "TTL": 300,
+      "ResourceRecords": [{"Value": "${vips[2]}"}]
+      }
+}]}
+EOF
+
+aws route53 change-resource-record-sets --hosted-zone-id "$hosted_zone_id" --change-batch file:///"${SHARED_DIR}"/dns-create.json
+
+echo "Creating batch file to destroy DNS records"
+cat > "${SHARED_DIR}"/dns-delete.json <<EOF
```

Comment:
:heart_eyes:

Reply:
Is the earlier creation atomic?  If not, we may want to make three separate create calls and then create an appropriate deletion JSON doc after each successful creation, to avoid leaking when we can create at least one, but not all, of the expected entries.

Reply:
> https://docs.aws.amazon.com/cli/latest/reference/route53/change-resource-record-sets.html

```
The request body contains a list of change items, known as a change batch. Change batches are considered transactional changes. When using the Amazon Route 53 API to change resource record sets, Route 53 either makes all or none of the changes in a change batch request. This ensures that Route 53 never partially implements the intended changes to the resource record sets in a hosted zone.
```

> Is the earlier creation atomic?

yes